js/src/volume.js

```diff
@@ -279,6 +282,20 @@ var ScatterView = widgets.WidgetView.extend( {
         _.mapObject(this.model.changedAttributes(), function(val, key){
             console.log("changed " +key)
             this.previous_values[key] = this.model.previous(key)
+            if (key == "index"){
+              pindex = this.previous_values["index"]
```

Comment:
Please use this.model.previous("index"). And if it is not defined, but x is a 2d array, we better assume it is zero. Or, maybe a better way, we don't allow sequence_index to be empty/null/None, and put it to 0 by default.